src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1464,6 +1464,185 @@ + (id)CHIPEncodeAndDecodeNSObject:(id)object
     return decodedData.GetDecodedObject();
 }
 
+template <typename DecodableEventType> class BufferedReadEventCallback final : public app::ReadClient::Callback {
```

Comment:
This is duplicating a lot of code that existing attribute-reading things have.  There was no way to share the code?

Also, why is this a template if it's only ever instantiated with one type for the template parameter?

Reply:
I agree your opinion that we should reduce code duplication. However in order to remove the duplication of code, lots of code changes are needed in the existing codes. So for now, I think that the separated codes can help to implement the read event function quickly and that code refactoring work can be done with more verification and discussion. The reason to use template is that I think to make the refactoring work easier.

Reply:
@seokhee-lee I appreciate your desire to add this quickly; are you able to commit to doing the needed refactoring afterward?